visualization/dashboard/src/MLIDashboard/Cohort.ts

```diff
@@ -0,0 +1,128 @@
+import { IFilter, FilterMethods } from "./Interfaces/IFilter";
+import { JointDataset } from "./JointDataset";
+import { ModelExplanationUtils } from "./ModelExplanationUtils";
+
+export class Cohort {
+    public static CohortKey: string = "Cohort";
+    private static _cohortIndex: number = 0;
+
+    public rowCount: number = 0;
+    private readonly cohortIndex: number;
+    private mutateCount: number = 0;
+    private _filteredData: Array<{[key: string]: number}>;
+    private _cachedAverageImportance: number[];
+    private _cachedTransposedLocalFeatureImportances: number[][];
+    private currentSortKey: string | undefined;
+    private currentSortReversed: boolean = false;
+    constructor(public name: string, private jointDataset: JointDataset, public filters: IFilter[] = []) {
```

Comment:
confused about this syntax, what does it mean to have both public and private parameters in the constructor?

Reply:
the passed in params can be stored as member properties on a class, can be public or private. the public fields could be made private and accessed by a getter/setter, but this is typescript :)